genshin/client/components/geetest/client.py

```diff
@@ -14,38 +16,36 @@
 __all__ = ["GeetestClient"]
 
 
-WEB_LOGIN_URL = yarl.URL("https://api-account-os.hoyolab.com/account/auth/api/webLoginByPassword")
+WEB_LOGIN_URL = yarl.URL("https://sg-public-api.hoyolab.com/account/ma-passport/api/webLoginByPassword")
 
 
 class GeetestClient(base.BaseClient):
     """Geetest client component."""
 
     async def login_with_geetest(
-        self,
-        account: str,
-        password: str,
-        mmt_key: str,
-        geetest: typing.Dict[str, str],
-        token_type: int = 0b111,
+        self, account: str, password: str, session_id: str, geetest: typing.Dict[str, str]
     ) -> typing.Mapping[str, str]:
         """Login with a password and a solved geetest.
 
         Token type is a bitfield of cookie_token, ltoken, stoken.
         """
-        payload = dict(
-            account=account,
-            password=geetest_utility.encrypt_geetest_password(password),
-            is_crypto="true",
-            source="account.mihoyo.com",
-            mmt_key=mmt_key,
-            token_type=token_type,
-        )
-        payload.update(geetest)
+        payload = {
+            "account": geetest_utility.encrypt_geetest_password(account),
+            "password": geetest_utility.encrypt_geetest_password(password),
+            "token_type": 6,
```

Comment:
Yup, Because mihoyo has split 2 endpoint `/webLoginByPassword` and `/appLoginBy Password`. You cannot get `SToken` with using endpoint `/webLoginByPassword `

If you want get `SToken`. You must request to `appLoginByPassword` to get `SToken`